### DISCUSSION TITLE
Rocket v0.5.0-rc.4

### DISCUSSION THREAD

**Discussion by SergioBenitez** [rwf2/Rocket#2637] [Announcements]
See the [CHANGELOG](https://github.com/SergioBenitez/Rocket/blob/v0.5.0-rc.4/CHANGELOG.md), [news article](https://rocket.rs/v0.5-rc/news/2023-11-01-version-0.5-rc.4/), [migration guide](https://rocket.rs/v0.5-rc/guide/upgrading/), and [FAQ](https://rocket.rs/v0.5-rc/guide/faq/) for more information.

<hr /><em>This discussion was created from the release <a href='https://github.com/SergioBenitez/Rocket/releases/tag/v0.5.0-rc.4'>Rocket v0.5.0-rc.4</a>.</em>

**Comment by MCOfficer** [comment]
It's probably a formality, but I believe you forgot to tag this one as pre-release.

**Reply by SergioBenitez** [reply]
> It's probably a formality, but I believe you forgot to tag this one as pre-release.

I purposefully tagged it as an "actual" release. GitHub implies, when it confirms your choice, that "pre-releases" are "not production ready," which isn't the case here, so I decided not to label it as such. 0.5-rc is definitely production ready. In a different world, each of these RCs could have easily been major releases.

**Reply by wezm** [reply]
> so I decided not to label it as such. 0.5-rc is definitely production ready. In a different world, each of these RCs could have easily been major releases.

Asking out of curiosity: given above why not make this one a non-rc release?

**Reply by SergioBenitez** [reply]
> Asking out of curiosity: given above why not make this one a non-rc release?

There's no _technical_ reason: the code itself is good to go. But there are several non-code pieces still left to complete for a 0.5 release, and I figured I'd release _something_ as opposed to delaying further. These include:

  * A consolidated CHANGELOG of all changes since 0.4.
  * A thorough review of the documentation, especially the upgrade guide.
  * A news article intended for wide distribution describing the release's major updates.
  * Adjustment of the code (version numbers, dates, etc.) for a stable release.

Additionally, it's my hope that I can properly launch the supporting foundation ([RWF2](https://rwf2.org)) alongside v0.5. That, of course, has a bevy of items that need to be completed for launch.

Nevertheless, I am diligently working towards all of this. The CHANGELOG should be completed shortly, at which point I'll continue to tackle the remaining tasks.

**Reply by wezm** [reply]
Thanks for the reply Sergio. Good luck with the non-code bits. Thanks again for Rocket. I have several websites on the previous rc that just chug along quietly.
